plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/semantics/context/SQLQueryRowsSourceContext.java

```diff
@@ -264,7 +266,10 @@ public SQLQuerySourcesInfoCollection getKnownSources() {
         return new SQLQuerySourcesInfoCollection() {
 
             private final Map<SQLQueryRowsSourceModel, SourceResolutionResult> resolutionResults =
-                new HashSet<>(rowsSources.values()).stream().collect(Collectors.toMap(s -> s.source, s -> s));
+                Stream.of(rowsSources.values(), dynamicTableSources.values())
+                  .flatMap(Collection::stream)
+                  .collect(Collectors.toSet())
+                  .stream().collect(Collectors.toMap(s -> s.source, Function.identity()));
```

Comment:
Is this supposed to remove duplicates? You can use `Stream#distinct`. Moreover, you don't even need to remove duplicates - converting to a map should inherently do that.